docs: added doc for alert-dialog (#891)
harness/canary: apps/portal/src/content/docs/components/alert-dialog.mdx

---
title: AlertDialog
description: AlertDialog component documentation
---

> The `AlertDialog` component provides a flexible UI element for displaying alert dialogs. It is composed of several subcomponents such as `AlertDialogHeader`, `AlertDialogFooter`, `AlertDialogTitle`, and `AlertDialogDescription` to offer a structured and customizable interface.

Reviewer: The subcomponents mentioned here aren't namespaced.
```suggestion
The `AlertDialog` component provides a flexible UI element for displaying alert dialogs. It is composed of several subcomponents such as `AlertDialog.Header`, `AlertDialog.Footer`, `AlertDialog.Title`, and `AlertDialog.Description` to offer a structured and customizable interface.
```

import { DocsPage } from "@/components/docs-page";

> import { AlertDialog } from "@harnessio/ui/components";
> import { Button } from "@harnessio/ui/components";

Reviewer: You don't need to import these unless they're being used in the MDX itself. The `ComponentExample` imports everything it needs to render and `ui` lib component.
```suggestion
```

<DocsPage.ComponentExample
  client:only
  code={`<AlertDialog.Root>
    <AlertDialog.Trigger asChild>
      <Button>Open Alert Dialog</Button>
    </AlertDialog.Trigger>
    <AlertDialog.Content>
      <AlertDialog.Header>
        <AlertDialog.Title>Alert Dialog Title</AlertDialog.Title>
      </AlertDialog.Header>

>       <AlertDialog.Description variant="secondary">
>         <p>Alert dialog content goes here. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.</p>

Reviewer: The `AlertDialog.Description` component wraps the content in a `p` tag already, so this triggers warnings in the console for being a paragraph inside a paragraph.
```suggestion
      <AlertDialog.Description variant="secondary">
        Alert dialog content goes here. Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
```

      </AlertDialog.Description>
      <AlertDialog.Footer>
        <AlertDialog.Cancel>Cancel</AlertDialog.Cancel>
        <AlertDialog.Action>Confirm</AlertDialog.Action>
      </AlertDialog.Footer>
    </AlertDialog.Content>
  </AlertDialog.Root>`}
/>

## Usage

```typescript jsx

> import { AlertDialog } from '@harnessio/ui/components'
> import { Button } from '@harnessio/ui/components'

Reviewer: ```suggestion
import { AlertDialog, Button } from '@harnessio/ui/components'
```

return (
  <AlertDialog.Root>
    <AlertDialog.Trigger asChild>
      <Button>Open Alert Dialog</Button>
    </AlertDialog.Trigger>
    <AlertDialog.Content>
      <AlertDialog.Header>
        <AlertDialog.Title>Alert Dialog Title</AlertDialog.Title>
      </AlertDialog.Header>
      <AlertDialog.Description variant="secondary">
        This is the alert dialog description.

>         <p>Alert dialog content goes here.</p>

Reviewer: Same here, the `Description` is a `p` anyway, so avoid anything that's not meant to be in a paragraph tag.
```suggestion
        Alert dialog content goes here.
```

      </AlertDialog.Description>
      <AlertDialog.Footer>

>         <AlertDialog.Cancel>Cancel</AlertDialog.Cancel>
>         <AlertDialog.Action>Confirm</AlertDialog.Action>

Reviewer: We should probably add suggestions here for how to react to cancellation and confirmation (especially that)
```suggestion
        <AlertDialog.Cancel>Cancel</AlertDialog.Cancel>
        <AlertDialog.Action onClick={handleConfirm}>
          Confirm
        </AlertDialog.Action>
```

Author: Sure, will add the same to propsTable as well

      </AlertDialog.Footer>
    </AlertDialog.Content>
  </AlertDialog.Root>
)
```

## Anatomy

> All parts of the AlertDialog component can be imported and composed as required.

Reviewer: ```suggestion
All parts of the `AlertDialog` component can be imported and composed as required.
```

```typescript jsx
<AlertDialog.Root>

>   <AlertDialog.Trigger>
>     <Button>Open Alert Dialog</Button>
>   </AlertDialog.Trigger>

Reviewer: Not sure which is better to suggest; use with `Button` and have the `Trigger` with `asChild`, or just have the trigger text 🤔 In this form, it will render as a `button` inside a `button`
```suggestion
  <AlertDialog.Trigger asChild>
    <Button>Open Alert Dialog</Button>
  </AlertDialog.Trigger>
```

Or
```suggestion
  <AlertDialog.Trigger>
    Open Alert Dialog
  </AlertDialog.Trigger>
```

Author: To keep it simple, I guess plain text would be good enough.

  <AlertDialog.Content>
    <AlertDialog.Header>
      <AlertDialog.Title>Alert Dialog Title</AlertDialog.Title>
    </AlertDialog.Header>
    <AlertDialog.Description variant="secondary">

>       This is the alert dialog description.
>       <p>Alert dialog content goes here.</p>

Reviewer: Again, only things that are allowed in a paragraph should be used in a `Description`
```suggestion
      This is the alert dialog description.
      Alert dialog content goes here.
```

    </AlertDialog.Description>
    <AlertDialog.Footer>
      <AlertDialog.Cancel>Cancel</AlertDialog.Cancel>
      <AlertDialog.Action>Confirm</AlertDialog.Action>
    </AlertDialog.Footer>
  </AlertDialog.Content>
</AlertDialog.Root>
```

## API Reference

### Root

The `Root` component serves as the main container for all alert dialog elements.

<DocsPage.PropsTable
  props={[
    {
      name: "children",
      description:
        "You can pass in your `AlertDialog.Trigger` and `AlertDialog.Content` as children",
      required: true,

>       value: "ReactNode",

Reviewer: Is `value` meant to be the type?

Author: Yeah mostly, just in case of enums, we mention the values directly

Reviewer: Not needed now, but we might want to change to `type` at some point to make it more obvious.

    },
  ]}
/>

### Trigger

The `Trigger` component is used to open the alert dialog.

```typescript jsx

> <AlertDialog.Trigger>
>   <Button>Open Alert Dialog</Button>
> </AlertDialog.Trigger>

Reviewer: This will render a `button` inside a `button`.
```suggestion
<AlertDialog.Trigger asChild>
  <Button>Open Alert Dialog</Button>
</AlertDialog.Trigger>
```

Or
```suggestion
<AlertDialog.Trigger>
  Open Alert Dialog
</AlertDialog.Trigger>
```

```

<DocsPage.PropsTable
  props={[
    {
      name: "children",
      description: "Content to display",
      required: true,
      value: "ReactNode",
    },
    {
      name: "asChild",

>       description: "If true, the component will be rendered as a child element",

Reviewer: This is a bit misleading since it's already rendered as a child of its parent. We might want to just use what Radix says about `asChild`.
```suggestion
      description: "Change the default rendered element for the one passed as a child, merging their props and behavior.",
```

```typescript jsx
<AlertDialog.Footer>
  <AlertDialog.Cancel>Cancel</AlertDialog.Cancel>
  <AlertDialog.Action>Confirm</AlertDialog.Action>
</AlertDialog.Footer>
```

<DocsPage.PropsTable
  props={[
    {
      name: "children",
      description: "Footer content",
      required: true,
      value: "ReactNode",
    },
    {
      name: "className",
      description: "Additional class names to apply to the alert dialog footer",
      required: false,
      value: "string",
    },
  ]}
/>

### Title

The `Title` component displays the title of the alert dialog.

```typescript jsx
<AlertDialog.Title>Alert Dialog Title</AlertDialog.Title>
```

<DocsPage.PropsTable
  props={[
    {
      name: "children",
      description: "Title content",
      required: true,
      value: "ReactNode",
    },
    {
      name: "className",
      description: "Additional class names to apply to the alert dialog title",
      required: false,
      value: "string",
    },
  ]}
/>

### Description

The `Description` component displays the description of the alert dialog.

```typescript jsx
<AlertDialog.Description variant="secondary">
  This is the alert dialog description.

>   <p>Alert dialog content goes here.</p>

Reviewer: ```suggestion
  Alert dialog content goes here.
```

</AlertDialog.Description>
```

<DocsPage.PropsTable
  props={[
    {
      name: "children",
      description: "Description content",
      required: true,
      value: "ReactNode",
    },
    {
      name: "className",
      description:
        "Additional class names to apply to the alert dialog description",
      required: false,
      value: "string",
    },
    {
      name: "variant",
      description: "Variant of the description",
      required: false,
      value: "'secondary' | 'quaternary'",
      defaultValue: "'secondary'",
    },
  ]}
/>

### Action

The `Action` component is used to define the action button of the alert dialog.

```typescript jsx
<AlertDialog.Action>Confirm</AlertDialog.Action>
```

> <DocsPage.PropsTable

Reviewer: Might be worth calling out `onClick` and `asChild` to show common usage to take an action when the button is clicked.

Author: I don't think we have onClick in AlertDialog.Action or Cancel,
will update the asChild and the example on how to handle actions

Reviewer: `AlertDialog.Action` and `AlertDialog.Cancel` both extend the `HTMLButtonElement` and render as a `button` element. Each of those has a few hundred props

https://github.com/user-attachments/assets/8c9598d1-8fc9-4584-af02-1b5ed5704830

  props={[
    {
      name: "children",
      description: "Action button content",
      required: true,
      value: "ReactNode",
    },
    {
      name: "className",
      description:
        "Additional class names to apply to the alert dialog action button",
      required: false,
      value: "string",
    },
  ]}
/>

### Cancel

The `Cancel` component is used to define the cancel button of the alert dialog.

```typescript jsx
<AlertDialog.Cancel>Cancel</AlertDialog.Cancel>
```

> <DocsPage.PropsTable

Reviewer: Same here, we don't demonstrate how to trigger anything when someone cancels or confirms.

### Root

The `Root` component serves as the main container for all alert dialog elements.

<DocsPage.PropsTable
  props={[
    {
      name: "children",
      description:
        "You can pass in your `AlertDialog.Trigger` and `AlertDialog.Content` as children",
      required: true,
      value: "ReactNode",
    },
  ]}
/>

### Trigger

The `Trigger` component is used to open the alert dialog.

```typescript jsx
<AlertDialog.Trigger>
  Open Alert Dialog
</AlertDialog.Trigger>
```

<DocsPage.PropsTable
  props={[
    {
      name: "children",
      description: "Content to display",
      required: true,
      value: "ReactNode",
    },
    {
      name: "asChild",
      description:
        "Change the default rendered element for the one passed as a child, merging their props and behavior.",
      required: false,
      value: "boolean",
    },
  ]}
/>

### Content

The `Content` component defines the main content area of the alert dialog.

```typescript jsx
<AlertDialog.Content>
  <AlertDialog.Header>
    <AlertDialog.Title>Alert Dialog Title</AlertDialog.Title>
  </AlertDialog.Header>
  <AlertDialog.Description variant="secondary">
    This is the alert dialog description.

>     <p>Alert dialog content goes here.</p>

Reviewer: ```suggestion
    Alert dialog content goes here.
```
